titiler/dependencies.py

```diff
@@ -44,201 +48,201 @@ async def request_hash(request: Request) -> str:
-        ),
-        height: Optional[int] = Query(None, description="Force output image height."),
-        width: Optional[int] = Query(None, description="Force output image width."),
-        nodata: Optional[Union[str, int, float]] = Query(
-            None, title="Nodata value", description="Overwrite internal Nodata value"
-        ),
-        rescale: Optional[str] = Query(
-            None,
-            title="Min/Max data Rescaling",
-            description="comma (',') delimited Min,Max bounds",
-        ),
-        color_formula: Optional[str] = Query(
-            None,
-            title="Color Formula",
-            description="rio-color formula (info: https://github.com/mapbox/rio-color)",
-        ),
-        color_map: Optional[ColorMapNames] = Query(
-            None, description="rio-tiler's colormap name"
-        ),
-        resampling_method: ResamplingNames = Query(
-            ResamplingNames.nearest, description="Resampling method."  # type: ignore
-        ),
-    ):
-        """Populate Imager Params."""
-        self.indexes = tuple(int(s) for s in re.findall(r"\d+", bidx)) if bidx else None
-        self.expression = expression
-        self.max_size = max_size
-        self.height = height
-        self.width = width
-        if self.width and self.height:
-            self.max_size = None
-        if nodata is not None:
-            nodata = numpy.nan if nodata == "nan" else float(nodata)
-        self.nodata = nodata
-        self.rescale = rescale
-        self.color_formula = color_formula
-        self.color_map = cmap.get(color_map.value) if color_map else None
-        kwargs = dict(request.query_params)
-        kwargs.pop("url", None)
-        kwargs.pop("format", None)
-        kwargs.pop("bidx", None)
-        kwargs.pop("expression", None)
-        kwargs.pop("nodata", None)
-        kwargs.pop("rescale", None)
-        kwargs.pop("color_formula", None)
-        kwargs.pop("color_map", None)
-        kwargs.pop("height", None)
-        kwargs.pop("width", None)
-        kwargs.pop("max_size", None)
-        kwargs.pop("assets", None)  # For STAC
-        self.kwargs = kwargs
-
-
-class CommonMetadataParams:
+def MosaicTMSParams(
+    TileMatrixSetId: MosaicTileMatrixSetNames = Query(
+        MosaicTileMatrixSetNames.WebMercatorQuad,  # type: ignore
+        description="TileMatrixSet Name (default: 'WebMercatorQuad')",
+    )
+) -> morecantile.TileMatrixSet:
+    """TileMatrixSet Dependency."""
+    return morecantile.tms.get(TileMatrixSetId.name)
+
+
+@dataclass
+class DefaultDependency:
+    """Dependency Base Class"""
+
+    kwargs: dict = field(init=False, default_factory=dict)
+
+
+@dataclass
+class PathParams(DefaultDependency):
+    """Create dataset path from args"""
+
+    url: str = Query(..., description="Dataset URL")
+    reader: Optional[Type[BaseReader]] = field(init=False, default=None)  # Placeholder
+
+    def __post_init__(self,):
```

Comment:
Using `__post_init__` here is serving the same purpose as defining a dependency (ex. [`TMSParams`](https://github.com/developmentseed/titiler/pull/79/files#diff-7e64e756c81e1d4282046f8561c647ebR51-R55)) just with a different pattern.  Would it be possible to use either one or the other everywhere you need to modify a request parameter after it is declared?

Reply:
Yeah except if we go ahead with `reader` defined as a scheme (e.g `landsat+...`) It's not really used right now but the Factory expects the PathParams class to have 2 properties (url+reader).